h/static/scripts/app-controller.coffee

```diff
@@ -53,9 +56,27 @@ module.exports = class AppController
     );
 
     identity.watch({
-      onlogin: (identity) -> $scope.auth.user = auth.userid(identity)
-      onlogout: -> $scope.auth.user = null
-      onready: -> $scope.auth.user ?= null
+      onlogin: (identity) ->
+        # Hide the account dialog
+        $scope.accountDialog.visible = false
+        # Update the current logged-in user information
+        userid = auth.userid(identity)
+        parsed = parseAccountID(userid)
+        angular.copy({
```

Comment:
In the Chrome extension where we're not using Angular I've started to use [Object.assign](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) or a polyfill for it for this kind of thing. We might want to consider doing the same in the main app in future.